### DISCUSSION TITLE
insert CSS from getStaticProps at end of <head>

### DISCUSSION THREAD

**Discussion by travis** [vercel/next.js#24845] [Help]
I'm pulling in a bit of CSS from a headless CMS in getStaticProps. I'd like to insert it at the end of `<head>` so that it overrides all other CSS in my application:

https://github.com/understory-garden/single-page-gate/blob/be664b299fe115ee00e400028ef6a61c41a4980a/pages/index.js#L24

I've investigated using `next/head` and `styled-jsx` for this, but from what I can tell, `next/head` doesn't let me ensure something will be added at the end of the `<head>`, and `styled-jsx` doesn't seem to allow me to use dynamically generated strings like this - the only dynamic support I could find there is for setting property values dynamically using static templates. 

Does anyone have any suggestions for how to make this work?

**Comment by bobarros** [comment]
Using just javascript it's not an option?

``` jsx
useEffect(() => {
  //////// Delete all custom css to make sure
  document.querySelectorAll('style,link[rel="stylesheet"]').forEach(item => item.remove());
  //////// load new one
  const head = document.getElementsByTagName('HEAD')[0];
  const link = document.createElement('link');
  link.rel = 'stylesheet';
  link.type = 'text/css';
  link.href = 'newcustom.css';
  // Append
  head.appendChild(link);
});
```

**Reply by travis** [reply]
ah, I've considered this and it's my current fallback yep! I'm concerned that this might not play nice with `next/head` and/or `styled-jsx` in the longer run, but if this is the best way to do it at the moment I'll give it a shot

**Reply by travis** [reply]
Ah to be clear - I don't want to delete the existing CSS - it's important that anything that is already in the page stays in place - I just want to add one more stylesheet at the end of the cascade to let the CMS user override the default CSS.
